internal/server/providers/claudeinstance/e2e_test.go

```
@@ -0,0 +1,307 @@
+	ID string `json:"id"`
+}
+
+// postQuery issues a GraphQL POST and decodes into graphqlResponse.
+// Failures are fatal — a broken transport invalidates the whole test.
+func postQuery(t *testing.T, url, query string) graphqlResponse {
+	t.Helper()
+	body, err := json.Marshal(map[string]string{"query": query})
+	if err != nil {
+		t.Fatalf("marshal query: %v", err)
+	}
+	req, err := http.NewRequest(http.MethodPost, url+"/graphql", strings.NewReader(string(body)))
```

golangci-lint (2.12.1)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)